File: 209. Minimum Size Subarray Sum/209. Minimum Size Subarray Sum.java
class Solution {

  public int minSubArrayLen(int target, int[] nums) {
    int minLength = Integer.MAX_VALUE;
    int currSum = 0;
    int left = 0;

    for (int i = 0; i < nums.length; i++) {
      currSum += nums[i];

      while (currSum >= target) {
        minLength = Math.min(minLength, i - left + 1);
        currSum -= nums[left];
        left++;
      }
    }

    return minLength < Integer.MAX_VALUE ? minLength : 0;
  }
}

File: 209. Minimum Size Subarray Sum/209. Minimum Size Subarray Sum.py
from typing import List


class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        min_length = float("inf")
        curr_sum = 0
        left = 0

        for i in range(len(nums)):
            curr_sum += nums[i]

            while curr_sum >= target:
                min_length = min(min_length, i - left + 1)
                curr_sum -= nums[left]
                left += 1

        return min_length if min_length < float("inf") else 0
